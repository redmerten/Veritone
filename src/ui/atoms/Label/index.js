// @flow

import { rem } from 'polished';
import styled, { css } from 'styled-components';

const Label = styled.label.attrs({ 'data-testid': 'atom-label' })`
  align-self: center;
  color: ${({ theme }) => theme.fonts.independence};
  font-family: 'roboto mono', monospace;
  font-size: ${rem(12)};
  font-weight: 500;
  letter-spacing: ${rem(0.43)};
  opacity: 0.75;
  padding-left: ${rem(4)};
  text-transform: uppercase;

  ${({ bold }) =>
    bold &&
    css`
      font-weight: 600;
    `}
`;

Label.displayName = 'Label';

export default Label;
