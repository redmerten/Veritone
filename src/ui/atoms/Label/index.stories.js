// @flow

import { storiesOf } from '@storybook/react';
import React from 'react';

import Label from '.';

storiesOf('Atoms/Label (form)', module).add('default', () => <Label htmlFor="name">Name</Label>);
