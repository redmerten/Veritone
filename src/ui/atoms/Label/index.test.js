// @flow

import React from 'react';

import Label from '.';

import { render } from 'utils/testing';

describe('atoms/Label', () => {
  it('matches its base snapshot', () => {
    const { baseElement } = render(<Label />);
    expect(baseElement).toMatchSnapshot();
  });

  it('renders label atom', () => {
    const { getByTestId } = render(<Label />);
    expect(getByTestId('atom-label')).toBeInTheDocument();
  });
});
