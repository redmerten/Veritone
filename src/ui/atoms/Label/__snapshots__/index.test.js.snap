// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Label matches its base snapshot 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-family: 'roboto mono',monospace;
  font-size: 0.75rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.026875rem;
  -moz-letter-spacing: 0.026875rem;
  -ms-letter-spacing: 0.026875rem;
  letter-spacing: 0.026875rem;
  opacity: 0.75;
  padding-left: 0.25rem;
  text-transform: uppercase;
}

<body>
  <div>
    <label
      class="c0"
      data-testid="atom-label"
    />
  </div>
</body>
`;
